### PR TITLE
Support scanning scope

### PR DESCRIPTION
**Pull request by rcohencyberarmor**
## PR Type:
Enhancement

___
## PR Description:
This pull request adds support for scanning scope in Kubescape. It introduces changes to the `getter/kscloudapi_test.go`, `datastructuresmethods.go`, `datastructuresmethods_test.go`, `prettyprinter.go`, `opaprocessor/utils.go`, and `opaprocessor/utils_test.go` files.

___
## PR Main Files Walkthrough:
- `getter/kscloudapi_test.go`: Updates the test for the `TestKSCloudAPI` function.
- `datastructuresmethods.go`: Adds a new function `getCloudType` to determine the cloud type based on the scan information. Also adds the `GetScanningScope` function to determine the scanning scope based on the scan information. Modifies the `Set` function to check if the framework is fit to the scanning scope.
- `datastructuresmethods_test.go`: Adds tests for the `isControlFitToScanScope` function.
- `prettyprinter.go`: Updates the footer message to include a link to more information about cluster scanning scope.
- `opaprocessor/utils.go`: Modifies the `ConvertFrameworksToPolicies` function to include the scanning scope in the `Set` function call.
- `opaprocessor/utils_test.go`: Adds tests for the `ConvertFrameworksToPolicies` function.

### CODE REVIEW

**Comment by rcohencyberarmor** [comment]
this PR is ready for review!

File: core/cautils/datastructuresmethods.go [hunk 1]
```diff
@@ -76,3 +79,73 @@ func isRuleKubescapeVersionCompatible(attributes map[string]interface{}, version
...
+	}
+	for i := range control.ScanningScope.Matches {
+		if isScanningScopeMatchToControlScope(scanScopeMatches, control.ScanningScope.Matches[i]) {
+			return true
+		}
+	}
+	return false
+}
+
+func isControlFitToScanning(control reporthandling.Control, scanScope reporthandling.ScanningScopeType) bool {
```
Comment (Daniel-GrunbergerCA) [inline core/cautils/datastructuresmethods.go:149]: why not call `isControlFitToScanScope` straight ?

File: core/cautils/datastructuresmethods.go [hunk 2]
```diff
@@ -15,15 +17,16 @@ func NewPolicies() *Policies {
 	}
 }
 
-func (policies *Policies) Set(frameworks []reporthandling.Framework, version string) {
+func (policies *Policies) Set(frameworks []reporthandling.Framework, version string, scanInfo *ScanInfo) {
```
Comment (Daniel-GrunbergerCA) [inline core/cautils/datastructuresmethods.go:20]: do we need to pass `*ScanInfo` or just `scanningContext`?

File: core/cautils/sliceutils.go [hunk 3]
```diff
@@ -0,0 +1,27 @@
+package cautils
+
+import "github.com/kubescape/opa-utils/reporthandling"
+
+func IsSubSliceScanningScopeType(haystack []reporthandling.ScanningScopeType, needle []reporthandling.ScanningScopeType) bool {
```
Comment (Daniel-GrunbergerCA) [inline core/cautils/sliceutils.go:5]: I am pretty sure there are more easily readable algorithms that can achieve that

File: core/cautils/datastructuresmethods.go [hunk 4]
```diff
@@ -76,3 +79,73 @@ func isRuleKubescapeVersionCompatible(attributes map[string]interface{}, version
...
+	}
+	return false, ""
+}
+
+func getScanningScope(scanInfo *ScanInfo) reporthandling.ScanningScopeType {
+	var result reporthandling.ScanningScopeType
+
+	switch scanInfo.GetScanningContext() {
+	case ContextCluster:
+		isCloud, cloudType := getCloudType(scanInfo)
```
Comment (Daniel-GrunbergerCA) [inline core/cautils/datastructuresmethods.go:101]: pls mind the linter comments

**Review by Daniel-GrunbergerCA — CHANGES_REQUESTED** [review]

File: core/cautils/datastructuresmethods_test.go [hunk 5]
```diff
@@ -0,0 +1,104 @@
...
+import (
+	"fmt"
+	"testing"
+
+	"github.com/kubescape/opa-utils/reporthandling"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func TestIsControlFitToScanning(t *testing.T) {
```
Comment (Daniel-GrunbergerCA) [inline core/cautils/datastructuresmethods_test.go:12]:
```suggestion
func TestIsControlFitToScanScope(t *testing.T) {
```

**Review by Daniel-GrunbergerCA — APPROVED** [review]

**Review by Daniel-GrunbergerCA — APPROVED** [review]

**Review by dwertent — APPROVED** [review]

**Pull request merged**